game.py

```diff
@@ -0,0 +1,304 @@
+                    player_sum  = [21],
+                    dealer_sum  = 21,
+                    blackjack   = 2)
+
+            # player gets 1.5 to 1 payout for Blackjack
+            return self.make_result_dict(
+                result      = 1.5 * bet, 
+                player_sum  = [21],
+                blackjack   = 1)
+
+        else:
+            # handle player decisions
+
+            # hit
+            while player_action == definitions.Actions.HIT and not split_aces:
+                # add card to player hand
+                player_hand.add(deck.random_card(True))
+
+                player_sum = self.player.hand_sum(player_hand)
+
+                # player busts on this draw and loses betting amount
+                if player_sum > 21:
+                
+                    # for testing
+                    if self.dev_mode:
+                        print("\n___Player's Cards After Bust___\n{}".format(player_hand))
+
+                    return self.make_result_dict(
+                        result      = -1 * bet, 
+                        player_sum  = [player_sum])
+
+                player_action = self.player.compute_play(player_hand, dealer_hand[0], split_aces)
+
+            # double
+            if player_action == definitions.Actions.DOUBLE:
+                double_down = 1
+
+                # add single card to player hand
+                player_hand.add(deck.random_card(True))
+                bet *= 2
+
+                # for testing
+                if self.dev_mode:
+                    print("\n___Player's Cards After Double___\n{}".format(player_hand))
+
+                player_sum = self.player.hand_sum(player_hand)
+
+                # player busts on this draw and loses betting amount
+                if player_sum > 21:
+                    return self.make_result_dict(
+                        result      = -1 * bet, 
+                        player_sum  = [player_sum],
+                        double_down = double_down)
+
+            # split
+            if player_action == definitions.Actions.SPLIT:
+                # make recursive calls and add the results of each hand
+                results = [
+                    self.game_result(bet, pydealer.Stack(cards = [player_hand[0]]), dealer_hand, deck, split=True),
+                    self.game_result(bet, pydealer.Stack(cards = [player_hand[1]]), dealer_hand, deck, split=True)
+                ]
+                
+                # play out the non-completed player hands
+                for i in range(2):
+                    if type(results[i]) is list:
+                        results[i] = self.compute_final_result(results[i][0], dealer_hand, deck, results[i][1], results[i][2])
+
+                # combine the two into a single dict
+                return self.make_result_dict(
+                    result      = results[0]["result"] + results[1]["result"], 
+                    player_sum  = results[0]["player_sum"] + results[1]["player_sum"],
```

Comment:
Wouldn't the Player's sum here possibly end up totaling over 21 here if both hands sums are being added?

Reply:
player_sum is a list, the + operation concatenates the two lists